pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/TextMatchTransformFunction.java

```diff
@@ -0,0 +1,100 @@
+
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.operator.ColumnContext;
+import org.apache.pinot.core.operator.blocks.ValueBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.segment.local.realtime.impl.invertedindex.NativeMutableTextIndex;
+import org.apache.pinot.segment.local.segment.index.readers.text.NativeTextIndexReader;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.segment.spi.index.reader.TextIndexReader;
+import org.roaringbitmap.buffer.MutableRoaringBitmap;
+
+
+/*
+ * This class implements TEXT_MATCH() as a transform function.
+ */
+public class TextMatchTransformFunction extends BaseTransformFunction {
+  public static final String FUNCTION_NAME = "textMatch";
+  private String _predicate;
+  private TextIndexReader _textIndexReader;
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, ColumnContext> columnContextMap) {
+    super.init(arguments, columnContextMap);
+
+    TransformFunction columnArg = arguments.get(0);
+    if (!(columnArg instanceof IdentifierTransformFunction && columnArg.getResultMetadata().isSingleValue())) {
+      throw new IllegalArgumentException(
+          "The first argument of TEXT_MATCH transform function must be a single-valued column");
+    }
+    String columnName = ((IdentifierTransformFunction) columnArg).getColumnName();
+    DataSource dataSource = columnContextMap.get(columnName).getDataSource();
+    if (dataSource == null) {
+      throw new IllegalArgumentException("Cannot apply TEXT_MATCH on column: " + columnName + " without text index");
+    }
+    TextIndexReader indexReader = dataSource.getTextIndex();
+    if (indexReader == null) {
+      throw new IllegalArgumentException("Cannot apply TEXT_MATCH on column: " + columnName + " without text index");
+    }
+    if (indexReader instanceof NativeTextIndexReader
+        || indexReader instanceof NativeMutableTextIndex) {
+      throw new UnsupportedOperationException(
+          "TEXT_MATCH is not supported on column: " + columnName + " with native text index");
+    }
+
+    TransformFunction predicate = arguments.get(1);
+    if (!(predicate instanceof LiteralTransformFunction && predicate.getResultMetadata().isSingleValue())) {
+      throw new IllegalArgumentException(
+          "The second argument of TEXT_MATCH transform function must be a single-valued string literal");
+    }
+
+    _predicate = ((LiteralTransformFunction) predicate).getStringLiteral();
+    _textIndexReader = indexReader;
+  }
+
+  public int[] transformToIntValuesSV(ValueBlock valueBlock) {
+    int length = valueBlock.getNumDocs();
+    initIntValuesSV(length);
+
+    int[] docIds = valueBlock.getDocIds();
+    MutableRoaringBitmap indexDocIds = _textIndexReader.getDocIds(_predicate);
+
```

Comment:
This approach is likely far from ideal for two reasons: 
- it iterates over docIds that could be large or `null` 
- doc ids could be a small chunk of a segment (e.g. 10k-big slice) while text index reader returns results for whole segment . Would it be better to cache the bitmap ?

Reply:
I don't remember how these slices work in SSE. I remember there are phases where the block is always as large as the whole segment and other phases where it is not.